mgc/discrim/base.py

```diff
@@ -0,0 +1,123 @@
+from abc import ABC, abstractmethod
+from sklearn.metrics import euclidean_distances
+from sklearn.utils import check_X_y
+import numpy as np
+
+
+class discriminabilityTest(ABC):
+    r"""
+    A base class for a Discriminability test.
+    
+    """
+    
+    
+    def __init__(self):
+        super().__init__()
+
+
+    def _statistic(self, X, Y, is_dist = False, remove_isolates=True, return_rdfs=False):
```

Comment:
again, move this to `OneSample` and move permutation test here

Reply:
test statistics calculation is same for both the tests, but the permutation test is different.